### DISCUSSION TITLE
OpenSSL (default) context cloning

### DISCUSSION THREAD

**Discussion by beldmit** [openssl/openssl#25778] [General Discussion]
Imagine an app, that loads the default openssl.conf and changes some libctx properties after that (based on their own configuration).

It makes sense to create a new context based on this changed properties and tune the freshly created context afterwards starting from the current settings of the app default context.

What API calls should be used for this purpose?

**Comment by mattcaswell** [comment]
I'm not entirely sure I understand what you are asking for.

If you simply want to create a new libctx with the default openssl.conf loaded into it then call `OSSL_LIB_CTX_new()` followed by `OSSL_LIB_CTX_load_config()` (with the config file specified as NULL - which loads the default config).

If you're after a `OSSL_LIB_CTX_dup` then we don't have such a thing - so that would be a feature request.

**Comment by beldmit** [comment]
Thank you, I raised #25780
